src/tags/index.js

```diff
@@ -95,19 +95,17 @@ module.exports = function(fractal){
 
                     if (token.withStack !== undefined) {
                         passedArguments = Twig.expression.parse.apply(this, [token.withStack, context]);
-
-                        _.forEach(passedArguments, function (value, name) {
-                            // It makes no sense to pass variables that are not
-                            // supported by the component.
-                            if (innerContext[name] === undefined) {
-                                return;
-                            }
-                            if (name.indexOf('attributes') > -1) {
-                                innerContext[name].merge(value);
-                            }
-                            else {
-                                innerContext[name] = value;
-                            }
+                        _.forEach(innerContext, function (value, name) {
+                          // Override default value only if an argument value is passed.
```

Comment:
As the condition is not obvious, but a relatively important key aspect in the variable handling, the code could use a comment educating about the situation and intended behavior:
```suggestion
                          // Override default value only if an argument value is passed.
                          // Ignore undefined variables, which may appear when rendering
                          // a component with dummy/faker data but without values for its
                          // child components, so that each component only generates its
                          // own dummy data.
```

Reply:
Indeed, thanks. 🙂